File: backend/schemas/birth_profile.py
from datetime import date
from datetime import time

from pydantic import BaseModel


class BirthProfileCreate(BaseModel):
    user_id: int

    date: date
    time: time

    place: str


class BirthProfileResponse(BaseModel):
    id: int

    user_id: int

    date: date
    time: time

    place: str

    latitude: float
    longitude: float

    timezone: str

    class Config:
        from_attributes = True

File: frontend/src/api/birth-profile.ts
import { api } from "./base";

export interface CreateBirthProfileRequest {
  user_id: number;
  date: string;
  time: string;
  place: string;
}

export interface BirthProfileResponse {
  id: number;
  user_id: number;

  date: string;
  time: string;

  place: string;

  latitude: number;
  longitude: number;

  timezone: string;
}

export async function createBirthProfile(
  payload: CreateBirthProfileRequest,
): Promise<BirthProfileResponse> {
  const response = await api.post(
    "/birth-profiles/",
    payload,
  );

  return response.data;
}

export async function getBirthProfile(
  userId: number,
): Promise<BirthProfileResponse> {
  const response = await api.get(
    `/birth-profiles/${userId}`,
  );

  return response.data;
}

export async function updateBirthProfile(
  userId: number,
  payload: CreateBirthProfileRequest,
): Promise<BirthProfileResponse> {
  const response = await api.put(
    `/birth-profiles/${userId}`,
    payload,
  );

  return response.data;
}
